lib/livebook/hubs/team_client.ex

```diff
@@ -534,4 +588,68 @@ defmodule Livebook.Hubs.TeamClient do
 
   defp nullify(""), do: nil
   defp nullify(value), do: value
+
+  defp deploy_apps(id, %{id: id} = deployment_group, derived_key) do
+    app_deployments =
+      for app_deployment <- deployment_group.app_deployments do
+        {:ok, app_deployment} = download_and_deploy(app_deployment, derived_key)
+        app_deployment
+      end
+
+    %{deployment_group | app_deployments: app_deployments}
+  end
+
+  defp deploy_apps(_, deployment_group, _), do: deployment_group
+
+  defp undeploy_apps(id, %{id: id} = deployment_group) do
+    deployment_group
+  end
+
+  defp undeploy_apps(_, deployment_group), do: deployment_group
+
+  defp download_and_deploy(%{file: nil} = app_deployment, _) do
+    app_deployment
+  end
+
+  defp download_and_deploy(app_deployment, derived_key) do
+    download_path = Path.join(System.tmp_dir!(), app_deployment.filename)
+    download_file = FileSystem.File.new(FileSystem.Local.new(), download_path)
```

Comment:
Is there a reason why we use `FileSystem.File` instead of using Elixir's `File` module directly?

Reply:
Yeah, we don't need to use the file abstraction for the zip files, they are always local!